###############################################################################################
# Pre-processing for data used to generate Figure 6 from Sayed, Kodama, Fan, et al. 2021 
# This script is STEP 1 of 3 

# Adapted from https://satijalab.org/seurat/v3.1/pbmc3k_tutorial.html
# by Li Fan 
###############################################################################################

library(Seurat)
library(ggplot2)
library(DoubletFinder)
library(SoupX)

#for loading Cell Ranger counts:
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_119/outs')
sc = autoEstCont(sc)
CV_NT_Ctrl_1.counts = adjustCounts(sc)
CV_NT_Ctrl_1 <- CreateSeuratObject(counts = CV_NT_Ctrl_1.counts, project = "LG72_119", min.cells = 3, min.features = 200)
CV_NT_Ctrl_1[["Genotype"]] = c('Human TREM2: +/+; P301S tau: -')
CV_NT_Ctrl_1[["Sample_Name"]] = c('CV_NT_Ctrl_1')
CV_NT_Ctrl_1[["Condition"]] = c('CV_NT_Ctrl')
CV_NT_Ctrl_1[["hTREM2"]] = c('CV')
CV_NT_Ctrl_1[["PS19"]] = c('NT')
CV_NT_Ctrl_1[["MK2206"]] = c('Ctrl')
rm(CV_NT_Ctrl_1.counts)

#vizualize QC metrics and filtering====
#mitochondrial transcripts - if the cell has high mitochondrial transcripts, it may signal a cell under stress/unhealthy
CV_NT_Ctrl_1[["percent.mt"]] <- PercentageFeatureSet(object = CV_NT_Ctrl_1, pattern = "^mt-") #recognize mitochondrial transcripts
all <- CV_NT_Ctrl_1
pdf("CV_NT_Ctrl_1_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
#plot of correlation between number of genes detected and number of transcripts detected - you generally want this to be 1:1
pdf("CV_NT_Ctrl_1_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
#initial filtering step - usually you want cells with detected gene counts over 300 and mitochondrial transcripts below 5%
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
#normalize counts=====
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("CV_NT_Ctrl_1_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("CV_NT_Ctrl_1_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
#Doublet finder (no ground-truth) - please reference https://github.com/chris-mcginnis-ucsf/DoubletFinder for more information on parameters====
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("CV_NT_Ctrl_1_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "CV_NT_Ctrl_1_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.099*13139) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
#doublet finder with different classification stringencies
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.005, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.005, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.005_1301", sct = FALSE)

#visualizing clusters and multiplet cells====
pdf("CV_NT_Ctrl_1_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("CV_NT_Ctrl_1_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.005_1301" #visualizing the singlet vs doublet cells
pdf("CV_NT_Ctrl_1_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
saveRDS(all,"CV_NT_Ctrl_1_after_doublet_detection.rds")

#processing singlets ====
#remove doublets
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"CV_NT_Ctrl_1_singlets.rds")
singlets<-readRDS("CV_NT_Ctrl_1_singlets.rds")
Idents(singlets) <- "seurat_clusters"
#normalization
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
#scaling the data
singlets <- ScaleData(object = singlets)
#perform and visualize PCA
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
#PC capture
pdf("CV_NT_Ctrl_1_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("CV_NT_Ctrl_1_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"CV_NT_Ctrl_1_singlets_PCA.rds")

###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_120/outs')
sc = autoEstCont(sc)
CV_NT_Ctrl_2.counts = adjustCounts(sc)
CV_NT_Ctrl_2 <- CreateSeuratObject(counts = CV_NT_Ctrl_2.counts, project = "LG72_120", min.cells = 3, min.features = 200)
CV_NT_Ctrl_2[["Genotype"]] = c('Human TREM2: +/+; P301S tau: -')
CV_NT_Ctrl_2[["Sample_Name"]] = c('CV_NT_Ctrl_2')
CV_NT_Ctrl_2[["Condition"]] = c('CV_NT_Ctrl')
CV_NT_Ctrl_2[["hTREM2"]] = c('CV')
CV_NT_Ctrl_2[["PS19"]] = c('NT')
CV_NT_Ctrl_2[["MK2206"]] = c('Ctrl')
rm(CV_NT_Ctrl_2.counts)
CV_NT_Ctrl_2[["percent.mt"]] <- PercentageFeatureSet(object = CV_NT_Ctrl_2, pattern = "^mt-") #recognize mitochondrial transcripts
all <- CV_NT_Ctrl_2
pdf("CV_NT_Ctrl_2_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("CV_NT_Ctrl_2_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("CV_NT_Ctrl_2_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("CV_NT_Ctrl_2_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("CV_NT_Ctrl_2_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "CV_NT_Ctrl_2_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.091*12122) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.005, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.005, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.005_1103", sct = FALSE)
pdf("CV_NT_Ctrl_2_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("CV_NT_Ctrl_2_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.005_1103" #visualizing the singlet vs doublet cells
pdf("CV_NT_Ctrl_2_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"CV_NT_Ctrl_2_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"CV_NT_Ctrl_2_singlets.rds")
#singlets<-readRDS("CV_NT_Ctrl_2_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("CV_NT_Ctrl_2_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("CV_NT_Ctrl_2_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"CV_NT_Ctrl_2_singlets_PCA.rds")

###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_127/outs')
sc = autoEstCont(sc)
CV_NT_Ctrl_3.counts = adjustCounts(sc)
CV_NT_Ctrl_3 <- CreateSeuratObject(counts = CV_NT_Ctrl_3.counts, project = "LG72_127", min.cells = 3, min.features = 200)
CV_NT_Ctrl_3[["Genotype"]] = c('Human TREM2: +/+; P301S tau: -')
CV_NT_Ctrl_3[["Sample_Name"]] = c('CV_NT_Ctrl_3')
CV_NT_Ctrl_3[["Condition"]] = c('CV_NT_Ctrl')
CV_NT_Ctrl_3[["hTREM2"]] = c('CV')
CV_NT_Ctrl_3[["PS19"]] = c('NT')
CV_NT_Ctrl_3[["MK2206"]] = c('Ctrl')
rm(CV_NT_Ctrl_3.counts)
CV_NT_Ctrl_3[["percent.mt"]] <- PercentageFeatureSet(object = CV_NT_Ctrl_3, pattern = "^mt-") #recognize mitochondrial transcripts
all <- CV_NT_Ctrl_3
pdf("CV_NT_Ctrl_3_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("CV_NT_Ctrl_3_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("CV_NT_Ctrl_3_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("CV_NT_Ctrl_3_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("CV_NT_Ctrl_3_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "CV_NT_Ctrl_3_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.091*12311) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.01, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.01, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.01_1120", sct = FALSE)
pdf("CV_NT_Ctrl_3_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("CV_NT_Ctrl_3_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.01_1120" #visualizing the singlet vs doublet cells
pdf("CV_NT_Ctrl_3_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"CV_NT_Ctrl_3_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"CV_NT_Ctrl_3_singlets.rds")
#singlets<-readRDS("CV_NT_Ctrl_3_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("CV_NT_Ctrl_3_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("CV_NT_Ctrl_3_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"CV_NT_Ctrl_3_singlets_PCA.rds")
###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_135/outs')
sc = autoEstCont(sc)
CV_NT_Ctrl_4.counts = adjustCounts(sc)
CV_NT_Ctrl_4 <- CreateSeuratObject(counts = CV_NT_Ctrl_4.counts, project = "LG72_135", min.cells = 3, min.features = 200)
CV_NT_Ctrl_4[["Genotype"]] = c('Human TREM2: +/+; P301S tau: -')
CV_NT_Ctrl_4[["Sample_Name"]] = c('CV_NT_Ctrl_4')
CV_NT_Ctrl_4[["Condition"]] = c('CV_NT_Ctrl')
CV_NT_Ctrl_4[["hTREM2"]] = c('CV')
CV_NT_Ctrl_4[["PS19"]] = c('NT')
CV_NT_Ctrl_4[["MK2206"]] = c('Ctrl')
rm(CV_NT_Ctrl_4.counts)
CV_NT_Ctrl_4[["percent.mt"]] <- PercentageFeatureSet(object = CV_NT_Ctrl_4, pattern = "^mt-") #recognize mitochondrial transcripts
all <- CV_NT_Ctrl_4
pdf("CV_NT_Ctrl_4_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("CV_NT_Ctrl_4_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("CV_NT_Ctrl_4_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("CV_NT_Ctrl_4_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("CV_NT_Ctrl_4_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "CV_NT_Ctrl_4_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.084*11609) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.06, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.06, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.06_975", sct = FALSE)
pdf("CV_NT_Ctrl_4_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("CV_NT_Ctrl_4_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.06_975" #visualizing the singlet vs doublet cells
pdf("CV_NT_Ctrl_4_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"CV_NT_Ctrl_4_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"CV_NT_Ctrl_4_singlets.rds")
#singlets<-readRDS("CV_NT_Ctrl_4_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("CV_NT_Ctrl_4_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("CV_NT_Ctrl_4_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"CV_NT_Ctrl_4_singlets_PCA.rds")
###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_80/outs')
sc = autoEstCont(sc)
CV_PS19_Ctrl_1.counts = adjustCounts(sc)
CV_PS19_Ctrl_1 <- CreateSeuratObject(counts = CV_PS19_Ctrl_1.counts, project = "LG72_80", min.cells = 3, min.features = 200)
CV_PS19_Ctrl_1[["Genotype"]] = c('Human TREM2: +/+; P301S tau: +')
CV_PS19_Ctrl_1[["Sample_Name"]] = c('CV_PS19_Ctrl_1')
CV_PS19_Ctrl_1[["Condition"]] = c('CV_PS19_Ctrl')
CV_PS19_Ctrl_1[["hTREM2"]] = c('CV')
CV_PS19_Ctrl_1[["PS19"]] = c('PS19')
CV_PS19_Ctrl_1[["MK2206"]] = c('Ctrl')
rm(CV_PS19_Ctrl_1.counts)
CV_PS19_Ctrl_1[["percent.mt"]] <- PercentageFeatureSet(object = CV_PS19_Ctrl_1, pattern = "^mt-") #recognize mitochondrial transcripts
all <- CV_PS19_Ctrl_1
pdf("CV_PS19_Ctrl_1_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("CV_PS19_Ctrl_1_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("CV_PS19_Ctrl_1_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("CV_PS19_Ctrl_1_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("CV_PS19_Ctrl_1_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "CV_PS19_Ctrl_1_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.084*11783) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.005, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.005, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.005_990", sct = FALSE)
pdf("CV_PS19_Ctrl_1_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("CV_PS19_Ctrl_1_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.005_990" #visualizing the singlet vs doublet cells
pdf("CV_PS19_Ctrl_1_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"CV_PS19_Ctrl_1_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"CV_PS19_Ctrl_1_singlets.rds")
#singlets<-readRDS("CV_PS19_Ctrl_1_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("CV_PS19_Ctrl_1_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("CV_PS19_Ctrl_1_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"CV_PS19_Ctrl_1_singlets_PCA.rds")
###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_81/outs')
sc = autoEstCont(sc)
CV_PS19_Ctrl_2.counts = adjustCounts(sc)
CV_PS19_Ctrl_2 <- CreateSeuratObject(counts = CV_PS19_Ctrl_2.counts, project = "LG72_81", min.cells = 3, min.features = 200)
CV_PS19_Ctrl_2[["Genotype"]] = c('Human TREM2: +/+; P301S tau: +')
CV_PS19_Ctrl_2[["Sample_Name"]] = c('CV_PS19_Ctrl_2')
CV_PS19_Ctrl_2[["Condition"]] = c('CV_PS19_Ctrl')
CV_PS19_Ctrl_2[["hTREM2"]] = c('CV')
CV_PS19_Ctrl_2[["PS19"]] = c('PS19')
CV_PS19_Ctrl_2[["MK2206"]] = c('Ctrl')
rm(CV_PS19_Ctrl_2.counts)
CV_PS19_Ctrl_2[["percent.mt"]] <- PercentageFeatureSet(object = CV_PS19_Ctrl_2, pattern = "^mt-") #recognize mitochondrial transcripts
all <- CV_PS19_Ctrl_2
pdf("CV_PS19_Ctrl_2_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("CV_PS19_Ctrl_2_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("CV_PS19_Ctrl_2_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("CV_PS19_Ctrl_2_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("CV_PS19_Ctrl_2_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "CV_PS19_Ctrl_2_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.084*11506) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.005, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.005, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.005_967", sct = FALSE)
pdf("CV_PS19_Ctrl_2_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("CV_PS19_Ctrl_2_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.005_967" #visualizing the singlet vs doublet cells
pdf("CV_PS19_Ctrl_2_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"CV_PS19_Ctrl_2_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"CV_PS19_Ctrl_2_singlets.rds")
#singlets<-readRDS("CV_PS19_Ctrl_2_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("CV_PS19_Ctrl_2_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("CV_PS19_Ctrl_2_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"CV_PS19_Ctrl_2_singlets_PCA.rds")
###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_131/outs')
sc = autoEstCont(sc)
CV_PS19_Ctrl_3.counts = adjustCounts(sc)
CV_PS19_Ctrl_3 <- CreateSeuratObject(counts = CV_PS19_Ctrl_3.counts, project = "LG72_131", min.cells = 3, min.features = 200)
CV_PS19_Ctrl_3[["Genotype"]] = c('Human TREM2: +/+; P301S tau: +')
CV_PS19_Ctrl_3[["Sample_Name"]] = c('CV_PS19_Ctrl_3')
CV_PS19_Ctrl_3[["Condition"]] = c('CV_PS19_Ctrl')
CV_PS19_Ctrl_3[["hTREM2"]] = c('CV')
CV_PS19_Ctrl_3[["PS19"]] = c('PS19')
CV_PS19_Ctrl_3[["MK2206"]] = c('Ctrl')
rm(CV_PS19_Ctrl_3.counts)
CV_PS19_Ctrl_3[["percent.mt"]] <- PercentageFeatureSet(object = CV_PS19_Ctrl_3, pattern = "^mt-") #recognize mitochondrial transcripts
all <- CV_PS19_Ctrl_3
pdf("CV_PS19_Ctrl_3_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("CV_PS19_Ctrl_3_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("CV_PS19_Ctrl_3_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("CV_PS19_Ctrl_3_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("CV_PS19_Ctrl_3_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "CV_PS19_Ctrl_3_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.091*12201) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.05, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.05, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.05_1110", sct = FALSE)
pdf("CV_PS19_Ctrl_3_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("CV_PS19_Ctrl_3_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.05_1110" #visualizing the singlet vs doublet cells
pdf("CV_PS19_Ctrl_3_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"CV_PS19_Ctrl_3_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"CV_PS19_Ctrl_3_singlets.rds")
#singlets<-readRDS("CV_PS19_Ctrl_3_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("CV_PS19_Ctrl_3_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("CV_PS19_Ctrl_3_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"CV_PS19_Ctrl_3_singlets_PCA.rds")
###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_133/outs')
sc = autoEstCont(sc)
CV_PS19_Ctrl_4.counts = adjustCounts(sc)
CV_PS19_Ctrl_4 <- CreateSeuratObject(counts = CV_PS19_Ctrl_4.counts, project = "LG72_133", min.cells = 3, min.features = 200)
CV_PS19_Ctrl_4[["Genotype"]] = c('Human TREM2: +/+; P301S tau: -')
CV_PS19_Ctrl_4[["Sample_Name"]] = c('CV_PS19_Ctrl_4')
CV_PS19_Ctrl_4[["Condition"]] = c('CV_PS19_Ctrl')
CV_PS19_Ctrl_4[["hTREM2"]] = c('CV')
CV_PS19_Ctrl_4[["PS19"]] = c('PS19')
CV_PS19_Ctrl_4[["MK2206"]] = c('Ctrl')
rm(CV_PS19_Ctrl_4.counts)
CV_PS19_Ctrl_4[["percent.mt"]] <- PercentageFeatureSet(object = CV_PS19_Ctrl_4, pattern = "^mt-") #recognize mitochondrial transcripts
all <- CV_PS19_Ctrl_4
pdf("CV_PS19_Ctrl_4_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("CV_PS19_Ctrl_4_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("CV_PS19_Ctrl_4_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("CV_PS19_Ctrl_4_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("CV_PS19_Ctrl_4_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "CV_PS19_Ctrl_4_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.084*11694) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.005, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.005, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.005_982", sct = FALSE)
pdf("CV_PS19_Ctrl_4_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("CV_PS19_Ctrl_4_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.005_982" #visualizing the singlet vs doublet cells
pdf("CV_PS19_Ctrl_4_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"CV_PS19_Ctrl_4_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"CV_PS19_Ctrl_4_singlets.rds")
#singlets<-readRDS("CV_PS19_Ctrl_4_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("CV_PS19_Ctrl_4_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("CV_PS19_Ctrl_4_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"CV_PS19_Ctrl_4_singlets_PCA.rds")
###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_71/outs')
sc = autoEstCont(sc)
R47H_NT_Ctrl_1.counts = adjustCounts(sc)
R47H_NT_Ctrl_1 <- CreateSeuratObject(counts = R47H_NT_Ctrl_1.counts, project = "LG72_71", min.cells = 3, min.features = 200)
R47H_NT_Ctrl_1[["Genotype"]] = c('Human TREM2: R47H/+; P301S tau: -')
R47H_NT_Ctrl_1[["Sample_Name"]] = c('R47H_NT_Ctrl_1')
R47H_NT_Ctrl_1[["Condition"]] = c('R47H_NT_Ctrl')
R47H_NT_Ctrl_1[["hTREM2"]] = c('R47H')
R47H_NT_Ctrl_1[["PS19"]] = c('NT')
R47H_NT_Ctrl_1[["MK2206"]] = c('Ctrl')
rm(R47H_NT_Ctrl_1.counts)
R47H_NT_Ctrl_1[["percent.mt"]] <- PercentageFeatureSet(object = R47H_NT_Ctrl_1, pattern = "^mt-") #recognize mitochondrial transcripts
all <- R47H_NT_Ctrl_1
pdf("R47H_NT_Ctrl_1_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("R47H_NT_Ctrl_1_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("R47H_NT_Ctrl_1_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_NT_Ctrl_1_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("R47H_NT_Ctrl_1_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "R47H_NT_Ctrl_1_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.091*12076) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.005, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.005, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.005_1099", sct = FALSE)
pdf("R47H_NT_Ctrl_1_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_NT_Ctrl_1_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.005_1099" #visualizing the singlet vs doublet cells
pdf("R47H_NT_Ctrl_1_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"R47H_NT_Ctrl_1_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"R47H_NT_Ctrl_1_singlets.rds")
#singlets<-readRDS("R47H_NT_Ctrl_1_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("R47H_NT_Ctrl_1_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("R47H_NT_Ctrl_1_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"R47H_NT_Ctrl_1_singlets_PCA.rds")
###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_102/outs')
sc = autoEstCont(sc)
R47H_NT_Ctrl_2.counts = adjustCounts(sc)
R47H_NT_Ctrl_2 <- CreateSeuratObject(counts = R47H_NT_Ctrl_2.counts, project = "LG72_102", min.cells = 3, min.features = 200)
R47H_NT_Ctrl_2[["Genotype"]] = c('Human TREM2: R47H/+; P301S tau: -')
R47H_NT_Ctrl_2[["Sample_Name"]] = c('R47H_NT_Ctrl_2')
R47H_NT_Ctrl_2[["Condition"]] = c('R47H_NT_Ctrl')
R47H_NT_Ctrl_2[["hTREM2"]] = c('R47H')
R47H_NT_Ctrl_2[["PS19"]] = c('NT')
R47H_NT_Ctrl_2[["MK2206"]] = c('Ctrl')
rm(R47H_NT_Ctrl_2.counts)
R47H_NT_Ctrl_2[["percent.mt"]] <- PercentageFeatureSet(object = R47H_NT_Ctrl_2, pattern = "^mt-") #recognize mitochondrial transcripts
all <- R47H_NT_Ctrl_2
pdf("R47H_NT_Ctrl_2_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("R47H_NT_Ctrl_2_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("R47H_NT_Ctrl_2_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_NT_Ctrl_2_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("R47H_NT_Ctrl_2_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "R47H_NT_Ctrl_2_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.076*10640) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.005, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.005, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.005_809", sct = FALSE)
pdf("R47H_NT_Ctrl_2_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_NT_Ctrl_2_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.005_809" #visualizing the singlet vs doublet cells
pdf("R47H_NT_Ctrl_2_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"R47H_NT_Ctrl_2_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"R47H_NT_Ctrl_2_singlets.rds")
#singlets<-readRDS("R47H_NT_Ctrl_2_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("R47H_NT_Ctrl_2_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("R47H_NT_Ctrl_2_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"R47H_NT_Ctrl_2_singlets_PCA.rds")
###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_132/outs')
sc = autoEstCont(sc)
R47H_NT_Ctrl_3.counts = adjustCounts(sc)
R47H_NT_Ctrl_3 <- CreateSeuratObject(counts = R47H_NT_Ctrl_3.counts, project = "LG72_132", min.cells = 3, min.features = 200)
R47H_NT_Ctrl_3[["Genotype"]] = c('Human TREM2: R47H/+; P301S tau: -')
R47H_NT_Ctrl_3[["Sample_Name"]] = c('R47H_NT_Ctrl_3')
R47H_NT_Ctrl_3[["Condition"]] = c('R47H_NT_Ctrl')
R47H_NT_Ctrl_3[["hTREM2"]] = c('R47H')
R47H_NT_Ctrl_3[["PS19"]] = c('NT')
R47H_NT_Ctrl_3[["MK2206"]] = c('Ctrl')
rm(R47H_NT_Ctrl_3.counts)
R47H_NT_Ctrl_3[["percent.mt"]] <- PercentageFeatureSet(object = R47H_NT_Ctrl_3, pattern = "^mt-") #recognize mitochondrial transcripts
all <- R47H_NT_Ctrl_3
pdf("R47H_NT_Ctrl_3_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("R47H_NT_Ctrl_3_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("R47H_NT_Ctrl_3_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_NT_Ctrl_3_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("R47H_NT_Ctrl_3_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "R47H_NT_Ctrl_3_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.084*11250) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.005, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.005, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.005_945", sct = FALSE)
pdf("R47H_NT_Ctrl_3_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_NT_Ctrl_3_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.005_945" #visualizing the singlet vs doublet cells
pdf("R47H_NT_Ctrl_3_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"R47H_NT_Ctrl_3_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"R47H_NT_Ctrl_3_singlets.rds")
#singlets<-readRDS("R47H_NT_Ctrl_3_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("R47H_NT_Ctrl_3_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("R47H_NT_Ctrl_3_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"R47H_NT_Ctrl_3_singlets_PCA.rds")
###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_78/outs')
sc = autoEstCont(sc)
R47H_NT_MK_1.counts = adjustCounts(sc)
R47H_NT_MK_1 <- CreateSeuratObject(counts = R47H_NT_MK_1.counts, project = "LG72_78", min.cells = 3, min.features = 200)
R47H_NT_MK_1[["Genotype"]] = c('Human TREM2: R47H/+; P301S tau: -')
R47H_NT_MK_1[["Sample_Name"]] = c('R47H_NT_MK_1')
R47H_NT_MK_1[["Condition"]] = c('R47H_NT_MK')
R47H_NT_MK_1[["hTREM2"]] = c('R47H')
R47H_NT_MK_1[["PS19"]] = c('NT')
R47H_NT_MK_1[["MK2206"]] = c('MK')
rm(R47H_NT_MK_1.counts)
R47H_NT_MK_1[["percent.mt"]] <- PercentageFeatureSet(object = R47H_NT_MK_1, pattern = "^mt-") #recognize mitochondrial transcripts
all <- R47H_NT_MK_1
pdf("R47H_NT_MK_1_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("R47H_NT_MK_1_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("R47H_NT_MK_1_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_NT_MK_1_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("R47H_NT_MK_1_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "R47H_NT_MK_1_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.076*10223) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.005, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.005, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.005_777", sct = FALSE)
pdf("R47H_NT_MK_1_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_NT_MK_1_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.005_777" #visualizing the singlet vs doublet cells
pdf("R47H_NT_MK_1_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"R47H_NT_MK_1_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"R47H_NT_MK_1_singlets.rds")
#singlets<-readRDS("R47H_NT_MK_1_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("R47H_NT_MK_1_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("R47H_NT_MK_1_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"R47H_NT_MK_1_singlets_PCA.rds")
###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_95/outs')
sc = autoEstCont(sc)
R47H_NT_MK_2.counts = adjustCounts(sc)
R47H_NT_MK_2 <- CreateSeuratObject(counts = R47H_NT_MK_2.counts, project = "LG72_95", min.cells = 3, min.features = 200)
R47H_NT_MK_2[["Genotype"]] = c('Human TREM2: R47H/+; P301S tau: -')
R47H_NT_MK_2[["Sample_Name"]] = c('R47H_NT_MK_2')
R47H_NT_MK_2[["Condition"]] = c('R47H_NT_MK')
R47H_NT_MK_2[["hTREM2"]] = c('R47H')
R47H_NT_MK_2[["PS19"]] = c('NT')
R47H_NT_MK_2[["MK2206"]] = c('MK')
rm(R47H_NT_MK_2.counts)
R47H_NT_MK_2[["percent.mt"]] <- PercentageFeatureSet(object = R47H_NT_MK_2, pattern = "^mt-") #recognize mitochondrial transcripts
all <- R47H_NT_MK_2
pdf("R47H_NT_MK_2_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("R47H_NT_MK_2_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("R47H_NT_MK_2_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_NT_MK_2_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("R47H_NT_MK_2_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "R47H_NT_MK_2_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.084*11410) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.005, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.005, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.005_958", sct = FALSE)
pdf("R47H_NT_MK_2_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_NT_MK_2_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.005_958" #visualizing the singlet vs doublet cells
pdf("R47H_NT_MK_2_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"R47H_NT_MK_2_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"R47H_NT_MK_2_singlets.rds")
#singlets<-readRDS("R47H_NT_MK_2_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("R47H_NT_MK_2_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("R47H_NT_MK_2_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"R47H_NT_MK_2_singlets_PCA.rds")
###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_146/outs')
sc = autoEstCont(sc)
R47H_NT_MK_3.counts = adjustCounts(sc)
R47H_NT_MK_3 <- CreateSeuratObject(counts = R47H_NT_MK_3.counts, project = "LG72_146", min.cells = 3, min.features = 200)
R47H_NT_MK_3[["Genotype"]] = c('Human TREM2: R47H/+; P301S tau: -')
R47H_NT_MK_3[["Sample_Name"]] = c('R47H_NT_MK_3')
R47H_NT_MK_3[["Condition"]] = c('R47H_NT_MK')
R47H_NT_MK_3[["hTREM2"]] = c('R47H')
R47H_NT_MK_3[["PS19"]] = c('NT')
R47H_NT_MK_3[["MK2206"]] = c('MK')
rm(R47H_NT_MK_3.counts)
R47H_NT_MK_3[["percent.mt"]] <- PercentageFeatureSet(object = R47H_NT_MK_3, pattern = "^mt-") #recognize mitochondrial transcripts
all <- R47H_NT_MK_3
pdf("R47H_NT_MK_3_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("R47H_NT_MK_3_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("R47H_NT_MK_3_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_NT_MK_3_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("R47H_NT_MK_3_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "R47H_NT_MK_3_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.084*11853) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.005, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.005, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.005_996", sct = FALSE)
pdf("R47H_NT_MK_3_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_NT_MK_3_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.005_996" #visualizing the singlet vs doublet cells
pdf("R47H_NT_MK_3_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"R47H_NT_MK_3_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"R47H_NT_MK_3_singlets.rds")
#singlets<-readRDS("R47H_NT_MK_3_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("R47H_NT_MK_3_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("R47H_NT_MK_3_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"R47H_NT_MK_3_singlets_PCA.rds")
###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_149/outs')
sc = autoEstCont(sc)
R47H_NT_MK_4.counts = adjustCounts(sc)
R47H_NT_MK_4 <- CreateSeuratObject(counts = R47H_NT_MK_4.counts, project = "LG72_149", min.cells = 3, min.features = 200)
R47H_NT_MK_4[["Genotype"]] = c('Human TREM2: R47H/+; P301S tau: -')
R47H_NT_MK_4[["Sample_Name"]] = c('R47H_NT_MK_4')
R47H_NT_MK_4[["Condition"]] = c('R47H_NT_MK')
R47H_NT_MK_4[["hTREM2"]] = c('R47H')
R47H_NT_MK_4[["PS19"]] = c('NT')
R47H_NT_MK_4[["MK2206"]] = c('MK')
rm(R47H_NT_MK_4.counts)
R47H_NT_MK_4[["percent.mt"]] <- PercentageFeatureSet(object = R47H_NT_MK_4, pattern = "^mt-") #recognize mitochondrial transcripts
all <- R47H_NT_MK_4
pdf("R47H_NT_MK_4_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("R47H_NT_MK_4_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("R47H_NT_MK_4_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_NT_MK_4_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("R47H_NT_MK_4_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "R47H_NT_MK_4_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.091*12918) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.005, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.005, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.005_1176", sct = FALSE)
pdf("R47H_NT_MK_4_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_NT_MK_4_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.005_1176" #visualizing the singlet vs doublet cells
pdf("R47H_NT_MK_4_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"R47H_NT_MK_4_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"R47H_NT_MK_4_singlets.rds")
#singlets<-readRDS("R47H_NT_MK_4_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("R47H_NT_MK_4_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("R47H_NT_MK_4_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"R47H_NT_MK_4_singlets_PCA.rds")
###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_70/outs')
sc = autoEstCont(sc)
R47H_PS19_Ctrl_1.counts = adjustCounts(sc)
R47H_PS19_Ctrl_1 <- CreateSeuratObject(counts = R47H_PS19_Ctrl_1.counts, project = "LG72_70", min.cells = 3, min.features = 200)
R47H_PS19_Ctrl_1[["Genotype"]] = c('Human TREM2: R47H/+; P301S tau: +')
R47H_PS19_Ctrl_1[["Sample_Name"]] = c('R47H_PS19_Ctrl_1')
R47H_PS19_Ctrl_1[["Condition"]] = c('R47H_PS19_Ctrl')
R47H_PS19_Ctrl_1[["hTREM2"]] = c('R47H')
R47H_PS19_Ctrl_1[["PS19"]] = c('PS19')
R47H_PS19_Ctrl_1[["MK2206"]] = c('Ctrl')
rm(R47H_PS19_Ctrl_1.counts)
R47H_PS19_Ctrl_1[["percent.mt"]] <- PercentageFeatureSet(object = R47H_PS19_Ctrl_1, pattern = "^mt-") #recognize mitochondrial transcripts
all <- R47H_PS19_Ctrl_1
pdf("R47H_PS19_Ctrl_1_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("R47H_PS19_Ctrl_1_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("R47H_PS19_Ctrl_1_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_PS19_Ctrl_1_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("R47H_PS19_Ctrl_1_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "R47H_PS19_Ctrl_1_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.069*9774) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.01, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.01, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.01_674", sct = FALSE)
pdf("R47H_PS19_Ctrl_1_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_PS19_Ctrl_1_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.01_674" #visualizing the singlet vs doublet cells
pdf("R47H_PS19_Ctrl_1_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"R47H_PS19_Ctrl_1_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"R47H_PS19_Ctrl_1_singlets.rds")
#singlets<-readRDS("R47H_PS19_Ctrl_1_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("R47H_PS19_Ctrl_1_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("R47H_PS19_Ctrl_1_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"R47H_PS19_Ctrl_1_singlets_PCA.rds")
###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_79/outs')
sc = autoEstCont(sc)
R47H_PS19_Ctrl_2.counts = adjustCounts(sc)
R47H_PS19_Ctrl_2 <- CreateSeuratObject(counts = R47H_PS19_Ctrl_2.counts, project = "LG72_79", min.cells = 3, min.features = 200)
R47H_PS19_Ctrl_2[["Genotype"]] = c('Human TREM2: R47H/+; P301S tau: +')
R47H_PS19_Ctrl_2[["Sample_Name"]] = c('R47H_PS19_Ctrl_2')
R47H_PS19_Ctrl_2[["Condition"]] = c('R47H_PS19_Ctrl')
R47H_PS19_Ctrl_2[["hTREM2"]] = c('R47H')
R47H_PS19_Ctrl_2[["PS19"]] = c('PS19')
R47H_PS19_Ctrl_2[["MK2206"]] = c('Ctrl')
rm(R47H_PS19_Ctrl_2.counts)
R47H_PS19_Ctrl_2[["percent.mt"]] <- PercentageFeatureSet(object = R47H_PS19_Ctrl_2, pattern = "^mt-") #recognize mitochondrial transcripts
all <- R47H_PS19_Ctrl_2
pdf("R47H_PS19_Ctrl_2_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("R47H_PS19_Ctrl_2_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("R47H_PS19_Ctrl_2_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_PS19_Ctrl_2_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("R47H_PS19_Ctrl_2_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "R47H_PS19_Ctrl_2_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.084*11098) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.01, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.01, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.01_932", sct = FALSE)
pdf("R47H_PS19_Ctrl_2_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_PS19_Ctrl_2_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.01_932" #visualizing the singlet vs doublet cells
pdf("R47H_PS19_Ctrl_2_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"R47H_PS19_Ctrl_2_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"R47H_PS19_Ctrl_2_singlets.rds")
#singlets<-readRDS("R47H_PS19_Ctrl_2_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("R47H_PS19_Ctrl_2_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("R47H_PS19_Ctrl_2_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"R47H_PS19_Ctrl_2_singlets_PCA.rds")
###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_105/outs')
sc = autoEstCont(sc)
R47H_PS19_Ctrl_3.counts = adjustCounts(sc)
R47H_PS19_Ctrl_3 <- CreateSeuratObject(counts = R47H_PS19_Ctrl_3.counts, project = "LG72_105", min.cells = 3, min.features = 200)
R47H_PS19_Ctrl_3[["Genotype"]] = c('Human TREM2: R47H/+; P301S tau: +')
R47H_PS19_Ctrl_3[["Sample_Name"]] = c('R47H_PS19_Ctrl_3')
R47H_PS19_Ctrl_3[["Condition"]] = c('R47H_PS19_Ctrl')
R47H_PS19_Ctrl_3[["hTREM2"]] = c('R47H')
R47H_PS19_Ctrl_3[["PS19"]] = c('PS19')
R47H_PS19_Ctrl_3[["MK2206"]] = c('Ctrl')
rm(R47H_PS19_Ctrl_3.counts)
R47H_PS19_Ctrl_3[["percent.mt"]] <- PercentageFeatureSet(object = R47H_PS19_Ctrl_3, pattern = "^mt-") #recognize mitochondrial transcripts
all <- R47H_PS19_Ctrl_3
pdf("R47H_PS19_Ctrl_3_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("R47H_PS19_Ctrl_3_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("R47H_PS19_Ctrl_3_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_PS19_Ctrl_3_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("R47H_PS19_Ctrl_3_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "R47H_PS19_Ctrl_3_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.084*11298) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.005, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.005, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.005_949", sct = FALSE)
pdf("R47H_PS19_Ctrl_3_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_PS19_Ctrl_3_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.005_949" #visualizing the singlet vs doublet cells
pdf("R47H_PS19_Ctrl_3_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"R47H_PS19_Ctrl_3_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"R47H_PS19_Ctrl_3_singlets.rds")
#singlets<-readRDS("R47H_PS19_Ctrl_3_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("R47H_PS19_Ctrl_3_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("R47H_PS19_Ctrl_3_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"R47H_PS19_Ctrl_3_singlets_PCA.rds")
###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_136/outs')
sc = autoEstCont(sc)
R47H_PS19_Ctrl_4.counts = adjustCounts(sc)
R47H_PS19_Ctrl_4 <- CreateSeuratObject(counts = R47H_PS19_Ctrl_4.counts, project = "LG72_136", min.cells = 3, min.features = 200)
R47H_PS19_Ctrl_4[["Genotype"]] = c('Human TREM2: R47H/+; P301S tau: +')
R47H_PS19_Ctrl_4[["Sample_Name"]] = c('R47H_PS19_Ctrl_4')
R47H_PS19_Ctrl_4[["Condition"]] = c('R47H_PS19_Ctrl')
R47H_PS19_Ctrl_4[["hTREM2"]] = c('R47H')
R47H_PS19_Ctrl_4[["PS19"]] = c('PS19')
R47H_PS19_Ctrl_4[["MK2206"]] = c('Ctrl')
rm(R47H_PS19_Ctrl_4.counts)
R47H_PS19_Ctrl_4[["percent.mt"]] <- PercentageFeatureSet(object = R47H_PS19_Ctrl_4, pattern = "^mt-") #recognize mitochondrial transcripts
all <- R47H_PS19_Ctrl_4
pdf("R47H_PS19_Ctrl_4_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("R47H_PS19_Ctrl_4_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("R47H_PS19_Ctrl_4_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_PS19_Ctrl_4_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("R47H_PS19_Ctrl_4_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "R47H_PS19_Ctrl_4_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.061*8835) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.02, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.02, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.02_539", sct = FALSE)
pdf("R47H_PS19_Ctrl_4_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_PS19_Ctrl_4_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.02_539" #visualizing the singlet vs doublet cells
pdf("R47H_PS19_Ctrl_4_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"R47H_PS19_Ctrl_4_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"R47H_PS19_Ctrl_4_singlets.rds")
#singlets<-readRDS("R47H_PS19_Ctrl_4_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("R47H_PS19_Ctrl_4_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("R47H_PS19_Ctrl_4_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"R47H_PS19_Ctrl_4_singlets_PCA.rds")
###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_111/outs')
sc = autoEstCont(sc)
R47H_PS19_MK_1.counts = adjustCounts(sc)
R47H_PS19_MK_1 <- CreateSeuratObject(counts = R47H_PS19_MK_1.counts, project = "LG72_111", min.cells = 3, min.features = 200)
R47H_PS19_MK_1[["Genotype"]] = c('Human TREM2: R47H/+; P301S tau: +')
R47H_PS19_MK_1[["Sample_Name"]] = c('R47H_PS19_MK_1')
R47H_PS19_MK_1[["Condition"]] = c('R47H_PS19_MK')
R47H_PS19_MK_1[["hTREM2"]] = c('R47H')
R47H_PS19_MK_1[["PS19"]] = c('PS19')
R47H_PS19_MK_1[["MK2206"]] = c('MK')
rm(R47H_PS19_MK_1.counts)
R47H_PS19_MK_1[["percent.mt"]] <- PercentageFeatureSet(object = R47H_PS19_MK_1, pattern = "^mt-") #recognize mitochondrial transcripts
all <- R47H_PS19_MK_1
pdf("R47H_PS19_MK_1_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("R47H_PS19_MK_1_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("R47H_PS19_MK_1_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_PS19_MK_1_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("R47H_PS19_MK_1_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "R47H_PS19_MK_1_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.091*12622) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.005, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.005, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.005_1149", sct = FALSE)
pdf("R47H_PS19_MK_1_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_PS19_MK_1_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.005_1149" #visualizing the singlet vs doublet cells
pdf("R47H_PS19_MK_1_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"R47H_PS19_MK_1_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"R47H_PS19_MK_1_singlets.rds")
#singlets<-readRDS("R47H_PS19_MK_1_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("R47H_PS19_MK_1_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("R47H_PS19_MK_1_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"R47H_PS19_MK_1_singlets_PCA.rds")
###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_112/outs')
sc = autoEstCont(sc)
R47H_PS19_MK_2.counts = adjustCounts(sc)
R47H_PS19_MK_2 <- CreateSeuratObject(counts = R47H_PS19_MK_2.counts, project = "LG72_112", min.cells = 3, min.features = 200)
R47H_PS19_MK_2[["Genotype"]] = c('Human TREM2: R47H/+; P301S tau: +')
R47H_PS19_MK_2[["Sample_Name"]] = c('R47H_PS19_MK_2')
R47H_PS19_MK_2[["Condition"]] = c('R47H_PS19_MK')
R47H_PS19_MK_2[["hTREM2"]] = c('R47H')
R47H_PS19_MK_2[["PS19"]] = c('PS19')
R47H_PS19_MK_2[["MK2206"]] = c('MK')
rm(R47H_PS19_MK_2.counts)
R47H_PS19_MK_2[["percent.mt"]] <- PercentageFeatureSet(object = R47H_PS19_MK_2, pattern = "^mt-") #recognize mitochondrial transcripts
all <- R47H_PS19_MK_2
pdf("R47H_PS19_MK_2_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("R47H_PS19_MK_2_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("R47H_PS19_MK_2_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_PS19_MK_2_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("R47H_PS19_MK_2_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "R47H_PS19_MK_2_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.091*12801) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.005, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.005, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.005_1165", sct = FALSE)
pdf("R47H_PS19_MK_2_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_PS19_MK_2_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.005_1165" #visualizing the singlet vs doublet cells
pdf("R47H_PS19_MK_2_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"R47H_PS19_MK_2_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"R47H_PS19_MK_2_singlets.rds")
#singlets<-readRDS("R47H_PS19_MK_2_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("R47H_PS19_MK_2_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("R47H_PS19_MK_2_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"R47H_PS19_MK_2_singlets_PCA.rds")
###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_130/outs')
sc = autoEstCont(sc)
R47H_PS19_MK_3.counts = adjustCounts(sc)
R47H_PS19_MK_3 <- CreateSeuratObject(counts = R47H_PS19_MK_3.counts, project = "LG72_130", min.cells = 3, min.features = 200)
R47H_PS19_MK_3[["Genotype"]] = c('Human TREM2: R47H/+; P301S tau: +')
R47H_PS19_MK_3[["Sample_Name"]] = c('R47H_PS19_MK_3')
R47H_PS19_MK_3[["Condition"]] = c('R47H_PS19_MK')
R47H_PS19_MK_3[["hTREM2"]] = c('R47H')
R47H_PS19_MK_3[["PS19"]] = c('PS19')
R47H_PS19_MK_3[["MK2206"]] = c('MK')
rm(R47H_PS19_MK_3.counts)
R47H_PS19_MK_3[["percent.mt"]] <- PercentageFeatureSet(object = R47H_PS19_MK_3, pattern = "^mt-") #recognize mitochondrial transcripts
all <- R47H_PS19_MK_3
pdf("R47H_PS19_MK_3_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("R47H_PS19_MK_3_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("R47H_PS19_MK_3_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_PS19_MK_3_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("R47H_PS19_MK_3_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "R47H_PS19_MK_3_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.084*11463) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.005, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.005, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.005_963", sct = FALSE)
pdf("R47H_PS19_MK_3_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_PS19_MK_3_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.005_963" #visualizing the singlet vs doublet cells
pdf("R47H_PS19_MK_3_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"R47H_PS19_MK_3_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"R47H_PS19_MK_3_singlets.rds")
#singlets<-readRDS("R47H_PS19_MK_3_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("R47H_PS19_MK_3_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("R47H_PS19_MK_3_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"R47H_PS19_MK_3_singlets_PCA.rds")
###############################################################################################
###############################################################################################
sc = load10X('/athena/ganlab/scratch/lif4001/LG72/cellranger_v3_nucleiRef_specifiyChemistry/LG72_142/outs')
sc = autoEstCont(sc)
R47H_PS19_MK_4.counts = adjustCounts(sc)
R47H_PS19_MK_4 <- CreateSeuratObject(counts = R47H_PS19_MK_4.counts, project = "LG72_142", min.cells = 3, min.features = 200)
R47H_PS19_MK_4[["Genotype"]] = c('Human TREM2: R47H/+; P301S tau: +')
R47H_PS19_MK_4[["Sample_Name"]] = c('R47H_PS19_MK_4')
R47H_PS19_MK_4[["Condition"]] = c('R47H_PS19_MK')
R47H_PS19_MK_4[["hTREM2"]] = c('R47H')
R47H_PS19_MK_4[["PS19"]] = c('PS19')
R47H_PS19_MK_4[["MK2206"]] = c('MK')
rm(R47H_PS19_MK_4.counts)
R47H_PS19_MK_4[["percent.mt"]] <- PercentageFeatureSet(object = R47H_PS19_MK_4, pattern = "^mt-") #recognize mitochondrial transcripts
all <- R47H_PS19_MK_4
pdf("R47H_PS19_MK_4_QC.pdf", width=12, height=4)
VlnPlot(object = all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size=0)
dev.off()
pdf("R47H_PS19_MK_4_FeatureScatter.pdf", width=12, height=4)
plot1 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot2 <- FeatureScatter(all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
CombinePlots(plots = list(plot1, plot2))
dev.off()
all <- subset(x = all, subset = nFeature_RNA > 300 & nFeature_RNA < 9000 & nCount_RNA < 50000 & percent.mt < 5)
all <- NormalizeData(all, normalization.method = "LogNormalize", scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
all <- ScaleData(object = all)
all <- RunPCA(object = all, features = rownames(x = all), verbose = FALSE)
pdf("R47H_PS19_MK_4_Elbow_1.pdf", width=8, height=6)
ElbowPlot(all)
dev.off()
all <- FindNeighbors(all, dims = 1:15)
all <- FindClusters(all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_PS19_MK_4_UMAP.pdf", width=8, height=6)
DimPlot(all, reduction = "umap", label = T)
dev.off()
sweep.pbmc <- paramSweep_v3(all,PCs=1:15,sct=FALSE)
sweep.stats_pbmc <- summarizeSweep(sweep.pbmc,GT=FALSE)
bcmvn_pbmc <- find.pK(sweep.stats_pbmc)
pdf("R47H_PS19_MK_4_ggplot_pK.pdf", width=18, height=6)
ggplot(bcmvn_pbmc, aes(x=bcmvn_pbmc$pK, y=bcmvn_pbmc$BCmetric))+geom_bar(stat="identity") #look for pK at the initial peak
dev.off()
write.csv(table(all$seurat_clusters), "R47H_PS19_MK_4_cell_counts_Condition.csv")
length(all@meta.data$seurat_clusters)
#homotypic doublet proportion estimate
annotations <- all@meta.data$seurat_clusters
homotypic.prop <- modelHomotypic(annotations)
nExp_poi <- round(0.076*10878) #estimate the number of multiplets you expect from the kit you are using - should give you percent expected based on number of nuclei inputs
nExp_poi.adj <- round(nExp_poi*(1-homotypic.prop))
all <- doubletFinder_v3(all, PCs=1:15, pN=0.25, pK=0.01, nExp=nExp_poi, reuse.pANN = FALSE,sct=FALSE)
all <- doubletFinder_v3(all, PCs = 1:15, pN = 0.25, pK = 0.01, nExp = nExp_poi.adj, reuse.pANN = "pANN_0.25_0.01_827", sct = FALSE)
pdf("R47H_PS19_MK_4_Elbow_2.pdf", width=8, height=6)
ElbowPlot(all,ndims=50)
dev.off()
all <- FindNeighbors(object = all, dims = 1:15)
all <- FindClusters(object = all, resolution = 0.1)
all <- RunUMAP(all, dims = 1:15)
pdf("R47H_PS19_MK_4_UMAP_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = T)
dev.off()
Idents(object = all) <- "DF.classifications_0.25_0.01_827" #visualizing the singlet vs doublet cells
pdf("R47H_PS19_MK_4_3_UMAP_singlets_doublets_2.pdf", width=8, height=6)
DimPlot(object = all, reduction = 'umap', label = F)
dev.off()
#saveRDS(all,"R47H_PS19_MK_4_after_doublet_detection.rds")
singlets <- subset(all, idents=c("Singlet"))
rm(all)
saveRDS(singlets,"R47H_PS19_MK_4_singlets.rds")
#singlets<-readRDS("R47H_PS19_MK_4_singlets.rds")
Idents(singlets) <- "seurat_clusters"
singlets <- NormalizeData(singlets, normalization.method = "LogNormalize", scale.factor = 10000)
singlets <- FindVariableFeatures(singlets, selection.method = "vst", nfeatures = 2000)
singlets <- ScaleData(object = singlets)
singlets <- RunPCA(object = singlets, features = rownames(x = singlets), verbose = FALSE)
pdf("R47H_PS19_MK_4_Elbow_after_processing.pdf", width=8, height=6)
ElbowPlot(singlets,ndims=50)
dev.off()
singlets <- FindNeighbors(object = singlets, dims = 1:15)
singlets <- FindClusters(object = singlets, resolution = 0.1)
singlets <- RunUMAP(object = singlets, dims = 1:15)
pdf("R47H_PS19_MK_4_UMAP_singlets_after_processing.pdf", width=8, height=6)
DimPlot(object = singlets, reduction = 'umap', label = T)
dev.off()
saveRDS(singlets,"R47H_PS19_MK_4_singlets_PCA.rds")
###############################################################################################










